evm/chains/base.py

```diff
@@ -606,12 +610,17 @@ def validate_block(self, block: BaseBlock) -> None:
         access to the ancestor blocks, this validation must occur at the Chain
         level.
         """
-        vm = self.get_vm()
-        vm.validate_seal(block.header)
-
+        self.validate_seal(block.header)
         self.validate_uncles(block)
         self.validate_gaslimit(block.header)
 
+    def validate_seal(self, header: BlockHeader) -> None:
+        """
+        Validate the seal on the given header.
+        """
+        vm = self.get_vm()
```

Comment:
I believe this should be `self.get_vm(header)` so that we validate according to the correct fork's rules.

Reply:
Actually, I'm already working in the area, so I'll take care of it.